presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -294,6 +295,7 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
 
         PlanNode root = new JoinNode(idAllocator.getNextId(),
                 JoinNode.Type.typeConvert(node.getType()),
+                DISTRIBUTED,
```

Comment:
No longer relevant with Optional method.  But this just needed a default because either way it would get rewritten.